### PR TITLE
Improvements on logging

### PR DESCRIPTION
**Pull request by tomoe**
- Add informative log messages for login (SUP-20)
- Add more info to http error log

### CODE REVIEW

**Comment by tomoe** [comment]
Hey @ryu25ish 

Could you take a look at this? with this patches, relogin sequence looks like this: 

```
2014-05-01 04:18:38.382 ERROR midonetclient.api_lib [-] Got http error(response={'status': '401', 'content-length': '45', 'access-control-expose-headers': 'Location', 'server': 'Apache-Coyote/1.1', 'date': 'Thu, 01 May 2014 04:18:38 GMT', 'access-control-allow-origin': '*', 'access-control-allow-methods': 'GET, POST, PUT, DELETE, OPTIONS', 'content-type': 'application/vnd.org.midonet.Error-v1+json;charset=UTF-8', 'www-authenticate': 'Basic  realm="Basic Midonet"'}, content='{"message":"Authentication error","code":401}') for request(uri=u'http://10.0.2.15:8080/midonet-api/bridges/897bb97b-fb3d-483e-907f-e8ffa9c60e6b', method='GET', body=None, query={},headers={'Content-Type': 'application/vnd.org.midonet.Bridge-v1+json', 'Accept': 'application/vnd.org.midonet.Bridge-v1+json', 'X-Auth-Token': '23a35b1b322d4e01ab9c4be9e8901ca3'}). Raising exception=<HTTPUnauthorized at 0x30a27d0 401 Unauthorized>
2014-05-01 04:18:38.383 INFO midonetclient.auth_lib [-] Got HTTPUnauthorized error, try logging in again
2014-05-01 04:18:38.383 INFO midonetclient.auth_lib [-] Logging in to MidoNet API server
```

File: src/midonetclient/api_lib.py [hunk 1]
```diff
@@ -87,6 +80,13 @@ def do_request(uri, method, body=None, query=None, headers=None):
 
     LOG.debug("do_request: response=%s | content=%s" % (response, content))
 
-    fail_if(is_http_error(response['status'], content))
+    if is_http_error(response['status']):
+	err = http_errors[response['status']](content)
+        LOG.error("Got http error(response=%r, content=%r) for "
```
Comment (ryu25ish) [inline src/midonetclient/api_lib.py:85]: Looks like there is an indentation problem here.  Could you try running pep8 on this?

Comment (tomoe) [inline src/midonetclient/api_lib.py:85]: Thanks for the catch.  Will do soon.
I was in a new vagrant VM without my vimrc, and TABs sneaked in...

**Comment by tomoe** [comment]
@ryu25ish Update the patch. Did pep8 and some manual tests.

**Pull request merged**